docs(topology): add context menu demo and documentation (#8683)
patternfly/patternfly-react: packages/react-topology/src/components/TopologyView/examples/TopologyContextMenu.md

---
id: Context Menu

> section: topology

Reviewer: sortValue: 23

---

Note: Topology lives in its own package at [`@patternfly/react-topology`](https://www.npmjs.com/package/@patternfly/react-topology)

import {
  ColaLayout,
  ContextMenuItem,
  ContextMenuSeparator,
  DefaultEdge,
  DefaultGroup,
  DefaultNode,
  EdgeStyle,
  GraphComponent,
  ModelKind,
  NodeShape,
  SELECTION_EVENT,
  Visualization,
  VisualizationProvider,
  VisualizationSurface,
  withContextMenu
} from '@patternfly/react-topology';

import './topology-example.css';

> import Icon1 from '@patternfly/react-icons/dist/esm/icons/regions-icon';

Reviewer: trying to keep the css import last in all the examples.

### Topology with context menus

> To add context menus to your node labels, you can use the `withContextMenu` utility when returning the component in the componentFactory, e.g.: `withContextMenu(() => contextMenu)(MyCustomNode)`.

Reviewer: ```suggestion
To add context menus to an element, you can use the `withContextMenu` utility when returning the component in the componentFactory, e.g.: `withContextMenu(() => contextMenu)(MyCustomNode)`.
```
